astropy/units/format/generic.py

```diff
@@ -529,18 +534,11 @@ def _parse_unit(cls, s: str, detailed_exception: bool = True) -> UnitBase:
     _regex_superscript: ClassVar[Pattern[str]] = re.compile(
         f"[{_superscripts}]?[{_superscripts[2:]}]+"
     )
-    _regex_deg: ClassVar[Pattern[str]] = re.compile("°([CF])?")
 
     @classmethod
     def _convert_superscript(cls, m: Match[str]) -> str:
         return f"({m.group().translate(cls._superscript_translations)})"
```

Comment:
Do you think it would be a good idea to refactor the code in a follow-up pull request to get rid of this method too?

Reply:
Not sure it can be captured as easily in an `if` statement, but in principle, yes. It would also be good to ensure mixes of superscript and regular digits do not work...